### PR TITLE
Allows NLS.parseMachineString to parse to an Amount

### PR DESCRIPTION
**Pull request by jmuscireum**

### CODE REVIEW

File: src/test/java/sirius/kernel/nls/NLSSpec.groovy [hunk 1]
```diff
@@ -168,6 +168,16 @@ class NLSSpec extends BaseSpecification {
         "0.1" | new BigDecimal("0.1")
     }
 
+    def "parseMachineString works for Amount"() {
+        expect:
+        NLS.parseMachineString(Amount.class, input) == output
+
+        where:
+        input | output
+        "0.1" | Amount.ONE.divideBy(Amount.TEN)
```
Comment (meg1502) [inline src/test/java/sirius/kernel/nls/NLSSpec.groovy]: Add a test for empty String and null which are way more intresting then these basic numbers.

File: src/main/java/sirius/kernel/nls/NLS.java [hunk 2]
```diff
@@ -1039,6 +1039,13 @@ private static String formatSpokenRecentDateWithTime(LocalDateTime givenDateTime
                 throw new IllegalArgumentException(fmtr("NLS.errInvalidDecimalNumber").set("value", value).format(), e);
             }
         }
+        if (Amount.class.equals(clazz)) {
```
Comment (meg1502) [inline src/main/java/sirius/kernel/nls/NLS.java]: isn't this super confusing that null or "" return null instead of Amount.NONE?

File: src/test/java/sirius/kernel/nls/NLSSpec.groovy (hunk 1 above)
Comment (jmuscireum) [reply src/test/java/sirius/kernel/nls/NLSSpec.groovy]: done

File: src/main/java/sirius/kernel/nls/NLS.java (hunk 2 above)
Comment (jmuscireum) [reply src/main/java/sirius/kernel/nls/NLS.java]: You are right, the alternative would be a workaround in sirius-search
